// yajsapi/dist/package/config.js
export const DEFAULTS = {
    engine: "vm",
    minMemGib: 0.5,
    minStorageGib: 2,
    minCpuThreads: 1,
    minCpuCores: 1,
    capabilities: [],
};
export var PackageFormat;
(function (PackageFormat) {
    PackageFormat["UNKNOWN"] = "";
    PackageFormat["GVMKIT_SQUASH"] = "gvmkit-squash";
})(PackageFormat || (PackageFormat = {}));
export class PackageConfig {
    packageFormat;
    imageHash;
    repoUrl;
    engine;
    minMemGib;
    minStorageGib;
    minCpuThreads;
    minCpuCores;
    capabilities;
    logger;
    constructor(options) {
        this.packageFormat = PackageFormat.GVMKIT_SQUASH;
        this.imageHash = options.imageHash;
        this.repoUrl = options.repoUrl;
        this.engine = options.engine || DEFAULTS.engine;
        this.minMemGib = options.minMemGib || DEFAULTS.minMemGib;
        this.minStorageGib = options.minStorageGib || DEFAULTS.minStorageGib;
        this.minCpuThreads = options.minCpuThreads || DEFAULTS.minCpuThreads;
        this.minCpuCores = options.minCpuCores || DEFAULTS.minCpuCores;
        this.capabilities = options.capabilities || DEFAULTS.capabilities;
        this.logger = options.logger;
    }
}
