// yajsapi/dist/events/events.js
export const EventType = "GolemEvent";
class CustomEvent extends Event {
    detail;
    constructor(type, data) {
        super(type, data);
        this.detail = data.detail;
    }
}
export class BaseEvent extends CustomEvent {
    constructor(data) {
        super(EventType, { detail: data });
    }
}
export class ComputationStarted extends BaseEvent {
}
export class ComputationFinished extends BaseEvent {
}
export class ComputationFailed extends BaseEvent {
}
export class TaskStarted extends BaseEvent {
}
export class TaskRedone extends BaseEvent {
}
export class TaskRejected extends BaseEvent {
}
export class TaskFinished extends BaseEvent {
}
export class AllocationCreated extends BaseEvent {
}
export class SubscriptionCreated extends BaseEvent {
}
export class SubscriptionFailed extends BaseEvent {
}
export class CollectFailed extends BaseEvent {
}
export class ProposalReceived extends BaseEvent {
}
export class ProposalRejected extends BaseEvent {
}
export class ProposalResponded extends BaseEvent {
}
export class ProposalConfirmed extends BaseEvent {
}
export class AgreementCreated extends BaseEvent {
}
export class AgreementConfirmed extends BaseEvent {
}
export class AgreementRejected extends BaseEvent {
}
export class AgreementTerminated extends BaseEvent {
}
export class InvoiceReceived extends BaseEvent {
}
export class DebitNoteReceived extends BaseEvent {
}
export class PaymentAccepted extends BaseEvent {
}
export class PaymentFailed extends BaseEvent {
}
export class ActivityCreated extends BaseEvent {
}
export class ActivityDestroyed extends BaseEvent {
}
export class ScriptSent extends BaseEvent {
}
export class ScriptExecuted extends BaseEvent {
}
