// yajsapi/dist/agreement/config.js
import { RequestorApi } from "ya-ts-client/dist/ya-market/api.js";
import { Configuration } from "ya-ts-client/dist/ya-market/index.js";
const DEFAULTS = {
    basePath: "http://127.0.0.1:7465",
    agreementRequestTimeout: 30000,
    agreementEventPoolingInterval: 5,
    agreementEventPoolingMaxEventsPerRequest: 100,
    agreementWaitingForProposalTimout: 10000,
    agreementWaitingForApprovalTimeout: 60,
};
export class AgreementConfig {
    options;
    agreementRequestTimeout;
    agreementWaitingForApprovalTimeout;
    api;
    logger;
    eventTarget;
    constructor(options) {
        this.options = options;
        const apiKey = options?.yagnaOptions?.apiKey || process.env.YAGNA_APPKEY;
        if (!apiKey)
            throw new Error("Api key not defined");
        const basePath = options?.yagnaOptions?.basePath || process.env.YAGNA_API_URL || DEFAULTS.basePath;
        const apiConfig = new Configuration({ apiKey, basePath: `${basePath}/market-api/v1`, accessToken: apiKey });
        this.api = new RequestorApi(apiConfig);
        this.agreementRequestTimeout = options?.agreementRequestTimeout || DEFAULTS.agreementRequestTimeout;
        this.agreementWaitingForApprovalTimeout =
            options?.agreementWaitingForApprovalTimeout || DEFAULTS.agreementWaitingForApprovalTimeout;
        this.logger = options?.logger;
        this.eventTarget = options?.eventTarget;
    }
}
export class AgreementServiceConfig extends AgreementConfig {
    agreementEventPoolingInterval;
    agreementEventPoolingMaxEventsPerRequest;
    agreementWaitingForProposalTimout;
    constructor(options) {
        super(options);
        this.agreementWaitingForProposalTimout =
            options?.agreementWaitingForProposalTimout || DEFAULTS.agreementWaitingForProposalTimout;
        this.agreementEventPoolingInterval =
            options?.agreementEventPoolingInterval || DEFAULTS.agreementEventPoolingInterval;
        this.agreementEventPoolingMaxEventsPerRequest =
            options?.agreementEventPoolingMaxEventsPerRequest || DEFAULTS.agreementEventPoolingMaxEventsPerRequest;
    }
}
