// ya-ts-client/dist/ya-activity/src/api/provider-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Yagna Activity API
 *  The Activity API can be perceived as controls which a Requestor-side application has to steer the execution of an Activity as specified in an Agreement which has been negotiated via the Market API/Protocol. This defines possible interactions between the Requestor application (via Activity API) and the generic components running on the Provider node, which host the Provider-side application code. The possible interactions imply a logical “execution environment” component, which is the host/container for the “payload” code. The “execution environment” is specified as an ExeUnit, with a generic interface via which a Provider node’s Activity Controller can operate the hosted code. It conforms with capability level 1 of the [Activity API specification] (https://docs.google.com/document/d/1BXaN32ediXdBHljEApmznSfbuudTU8TmvOmHKl0gmQM).
 *
 * The version of the OpenAPI document: 1.6.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProviderApi = exports.ProviderApiFactory = exports.ProviderApiFp = exports.ProviderApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../../common");
// @ts-ignore
const base_1 = require("../../base");
/**
 * ProviderApi - axios parameter creator
 * @export
 */
exports.ProviderApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Fetch Requestor command events.
         * @param {string} [appSessionId] A correlation/session identifier used for querying events related to an action where this appSessionId has been specified
         * @param {string} [afterTimestamp] Apply only to records created later than the specified timestamp
         * @param {number} [timeout] Timeout used in long-polling calls (in seconds). How many seconds server should wait for response containing new events (&#x60;0.0&#x60; means it should return immediately if there are no events)
         * @param {number} [maxEvents] Maximum number of events that server should return at once.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        collectActivityEvents: (appSessionId, afterTimestamp, timeout, maxEvents, options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/events`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication app_key required
            // http bearer authentication required
            yield common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (appSessionId !== undefined) {
                localVarQueryParameter['appSessionId'] = appSessionId;
            }
            if (afterTimestamp !== undefined) {
                localVarQueryParameter['afterTimestamp'] = (afterTimestamp instanceof Date) ?
                    afterTimestamp.toISOString() :
                    afterTimestamp;
            }
            if (timeout !== undefined) {
                localVarQueryParameter['timeout'] = timeout;
            }
            if (maxEvents !== undefined) {
                localVarQueryParameter['maxEvents'] = maxEvents;
            }
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get state of specified Activity.
         * @param {string} activityId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getActivityState: (activityId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'activityId' is not null or undefined
            common_1.assertParamExists('getActivityState', 'activityId', activityId);
            const localVarPath = `/activity/{activityId}/state`
                .replace(`{${"activityId"}}`, encodeURIComponent(String(activityId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication app_key required
            // http bearer authentication required
            yield common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get usage of specified Activity.
         * @param {string} activityId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getActivityUsage: (activityId, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'activityId' is not null or undefined
            common_1.assertParamExists('getActivityUsage', 'activityId', activityId);
            const localVarPath = `/activity/{activityId}/usage`
                .replace(`{${"activityId"}}`, encodeURIComponent(String(activityId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication app_key required
            // http bearer authentication required
            yield common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Set state of specified Activity.
         * @param {string} activityId
         * @param {ActivityState} activityState
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setActivityState: (activityId, activityState, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'activityId' is not null or undefined
            common_1.assertParamExists('setActivityState', 'activityId', activityId);
            // verify required parameter 'activityState' is not null or undefined
            common_1.assertParamExists('setActivityState', 'activityState', activityState);
            const localVarPath = `/activity/{activityId}/state`
                .replace(`{${"activityId"}}`, encodeURIComponent(String(activityId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'PUT' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication app_key required
            // http bearer authentication required
            yield common_1.setBearerAuthToObject(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            common_1.setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = common_1.serializeDataIfNeeded(activityState, localVarRequestOptions, configuration);
            return {
                url: common_1.toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
/**
 * ProviderApi - functional programming interface
 * @export
 */
exports.ProviderApiFp = function (configuration) {
    const localVarAxiosParamCreator = exports.ProviderApiAxiosParamCreator(configuration);
    return {
        /**
         *
         * @summary Fetch Requestor command events.
         * @param {string} [appSessionId] A correlation/session identifier used for querying events related to an action where this appSessionId has been specified
         * @param {string} [afterTimestamp] Apply only to records created later than the specified timestamp
         * @param {number} [timeout] Timeout used in long-polling calls (in seconds). How many seconds server should wait for response containing new events (&#x60;0.0&#x60; means it should return immediately if there are no events)
         * @param {number} [maxEvents] Maximum number of events that server should return at once.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        collectActivityEvents(appSessionId, afterTimestamp, timeout, maxEvents, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.collectActivityEvents(appSessionId, afterTimestamp, timeout, maxEvents, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @summary Get state of specified Activity.
         * @param {string} activityId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getActivityState(activityId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getActivityState(activityId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @summary Get usage of specified Activity.
         * @param {string} activityId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getActivityUsage(activityId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getActivityUsage(activityId, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         *
         * @summary Set state of specified Activity.
         * @param {string} activityId
         * @param {ActivityState} activityState
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setActivityState(activityId, activityState, options) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.setActivityState(activityId, activityState, options);
                return common_1.createRequestFunction(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
    };
};
/**
 * ProviderApi - factory interface
 * @export
 */
exports.ProviderApiFactory = function (configuration, basePath, axios) {
    const localVarFp = exports.ProviderApiFp(configuration);
    return {
        /**
         *
         * @summary Fetch Requestor command events.
         * @param {string} [appSessionId] A correlation/session identifier used for querying events related to an action where this appSessionId has been specified
         * @param {string} [afterTimestamp] Apply only to records created later than the specified timestamp
         * @param {number} [timeout] Timeout used in long-polling calls (in seconds). How many seconds server should wait for response containing new events (&#x60;0.0&#x60; means it should return immediately if there are no events)
         * @param {number} [maxEvents] Maximum number of events that server should return at once.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        collectActivityEvents(appSessionId, afterTimestamp, timeout, maxEvents, options) {
            return localVarFp.collectActivityEvents(appSessionId, afterTimestamp, timeout, maxEvents, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get state of specified Activity.
         * @param {string} activityId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getActivityState(activityId, options) {
            return localVarFp.getActivityState(activityId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get usage of specified Activity.
         * @param {string} activityId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getActivityUsage(activityId, options) {
            return localVarFp.getActivityUsage(activityId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Set state of specified Activity.
         * @param {string} activityId
         * @param {ActivityState} activityState
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setActivityState(activityId, activityState, options) {
            return localVarFp.setActivityState(activityId, activityState, options).then((request) => request(axios, basePath));
        },
    };
};
/**
 * ProviderApi - object-oriented interface
 * @export
 * @class ProviderApi
 * @extends {BaseAPI}
 */
class ProviderApi extends base_1.BaseAPI {
    /**
     *
     * @summary Fetch Requestor command events.
     * @param {string} [appSessionId] A correlation/session identifier used for querying events related to an action where this appSessionId has been specified
     * @param {string} [afterTimestamp] Apply only to records created later than the specified timestamp
     * @param {number} [timeout] Timeout used in long-polling calls (in seconds). How many seconds server should wait for response containing new events (&#x60;0.0&#x60; means it should return immediately if there are no events)
     * @param {number} [maxEvents] Maximum number of events that server should return at once.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProviderApi
     */
    collectActivityEvents(appSessionId, afterTimestamp, timeout, maxEvents, options) {
        return exports.ProviderApiFp(this.configuration).collectActivityEvents(appSessionId, afterTimestamp, timeout, maxEvents, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get state of specified Activity.
     * @param {string} activityId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProviderApi
     */
    getActivityState(activityId, options) {
        return exports.ProviderApiFp(this.configuration).getActivityState(activityId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get usage of specified Activity.
     * @param {string} activityId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProviderApi
     */
    getActivityUsage(activityId, options) {
        return exports.ProviderApiFp(this.configuration).getActivityUsage(activityId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Set state of specified Activity.
     * @param {string} activityId
     * @param {ActivityState} activityState
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProviderApi
     */
    setActivityState(activityId, activityState, options) {
        return exports.ProviderApiFp(this.configuration).setActivityState(activityId, activityState, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.ProviderApi = ProviderApi;
